crates/e2e-testing/src/spin.rs

```diff
@@ -37,37 +35,33 @@ pub fn new_app<'a>(
         cmd.append(&mut args);
     }
 
-    return utils::run(cmd, Some(basedir.as_str()), None);
+    let output = utils::run(&cmd, Some(&basedir), None)?;
+    utils::assert_success(&output);
+    Ok(output)
 }
 
 pub fn install_plugins(plugins: Vec<&str>) -> Result<Output> {
     // lock mutex to ensure one install_plugins runs at a time
-    let x = INSTALLING_PLUGINS_MUTEX.lock().unwrap();
+    let _lock = INSTALLING_PLUGINS_MUTEX.lock().unwrap();
 
-    let mut output = utils::run(vec!["spin", "plugin", "update"], None, None)?;
+    let mut output = utils::run(&["spin", "plugin", "update"], None, None)?;
+    utils::assert_success(&output);
 
     for plugin in plugins {
-        output = utils::run(
-            vec!["spin", "plugin", "install", plugin, "--yes"],
-            None,
-            None,
-        )?;
+        output = utils::run(&["spin", "plugin", "install", plugin, "--yes"], None, None)?;
+        utils::assert_success(&output);
     }
 
-    //this ensure we have mutex lock until here
-    drop(x);
```

Comment:
As far as I could tell this drop call is not necessary (the mutex lock will be dropped right after this at the end of the function.

Reply:
@itowlson suggested to use explicit drops for better readability.

Reply:
Hmm... I understand that when the drop happens early, and you want to show that the lock has been given up before doing more work, but the drop will happen right after this... This strikes me as fairly unidiomatic.